### ISSUE TITLE
SetHostInfo being called but BoltNetwork.SessionList stays at 0

### ISSUE THREAD

**Issue by DMeville** [BoltEngine/Bolt-Tracker#62]
###How to reproduce issue
1. Install latest bolt.
2.  Install Photon and Photon Samples
3. Use this (slightly modified) version of stanchion's "Lan autoconnect hack" from the discord: https://pastebin.com/HzzT2dTi 
edited so that it calls SetHostInfo automatically after bolt has started.
4. Start one client, start one server.  The server will automatically set host info and publish their info which *should* show up in BoltNetwork.SessionList 

###Expected Behavior
Client should detect the published host info in BoltNetwork.SessionList

###Actual behavior
Client *sometimes* has an empty BoltNetwork.SessionList even after waiting minutes after SetHostInfo has been called on the server.

###Additional Notes
Reproducing this behavior was difficult as it seems very inconsistent and I couldn't narrow down exactly why things fail. It happens very infrequently, but has popped up more than a few times while working on my project.  In most cases, the client detects the servers session in the session list in about 10 seconds.  Every 30 tests or so the server reports successfully calling SetHostInfo but the client never detects this new session.  I've tried waiting for 30 minutes to see if the session was just taking a while to propagate to the client's server list, but when the server doesn't show up in the first 30 seconds it doesn't show up at all.  I've tried restarting the client while keeping the server running, but the fresh client doesn't detect the session either. The server logs in my tests confirm SetHostInfo was called and there are no warnings, errors, or exceptions thrown anywhere.

Once the error has been reproduced (session list has not received the new session after waiting for 5 minutes), manually calling SetHostInfo on the server again results in the session list properly showing the server, responding quickly and updating the session list within 15 seconds.

This leads me to believe that SetHostInfo isn't 100% reliable, and sometimes the host info fails to reach the photon servers for some reason (dropped packet, internally blocked by *something* like not being connected to photon, idk?) and then fails silently without the server knowing if they need to SetHostInfo and try again to ensure the session is published correctly to clients.

Calling SetHostInfo redundantly a few times _seems_ to ensure that the server has been published to the server list properly, but as this issue is inconsistent and difficult to reproduce I'm not 100% sure it's fix.  The issue has not yet happened to me after implementing this fix.

Eg, Wrapping SetHostInfo with something like this:
```
public int setHostCount = 5; //will call SetHostInfo 5 times, once every 5 seconds
public void SetServerHostInfo() {
    BoltNetwork.SetHostInfo(/*stuff*/);
    SetHostCount--;
    if(setHostCount > 0) Invoke("SetServerHostInfo", 5f);
}
```

This issue does not seem to be related to #60 as no NullRefException is thrown on the server.

###Configuration
Unity Version: 2017.3.0p1
Bolt Version 1.1.0.5
Operating System: Windows 10
Network Configuration:

**Comment by ramonmelo** [comment]
Hello,

I ran several tests, using local and external machines, and all the times the sessions appears and disappears as expected. Creating several server and client instances, all worked fine.

Can you test your implementation with the attached code ? 

- `PhotonPoller.cs`: Replacement with the latest version of the middlewere between Bolt and Photon Cloud. Make a backup and put it at `Assets\photon\PhotonPlatform`.
- `PhotonTokens.cs`: Some Tokens used by the demo code.
- `PhotonInit.cs`: The basic Menu to start as Server or Client. This code will call `SetHostInfo` after Bolt finishes the start process.

You will need the update the code and select your *InGame* scene to be loaded when the `Server` starts, inside the `PhotonInit.cs`, line `49`.

[BoltPhoton.zip](https://github.com/BoltEngine/Bolt-Tracker/files/1951978/BoltPhoton.zip)

**Comment by DMeville** [comment]
Sorry about the delay!

Before trying with your code, I ran some tests again today to ensure that my issue wasn't caused by poor internet connection the day I submitted the issue or something.  This test confirmed that my original project behaved the same as outlined above - still broken (although it took 71 connect attempts for it to break)

Because of the hard to reproduce nature of this issue, I created a test project that is all automated.  It launches a local server and tries to connect to it, then shuts down and repeats if it was successful, logging the time it takes between launching the local server and detecting it on the clients session list.    This is so I can leave it running and go do something else with out having to sit there and press "Connect" every 10 seconds to see if it works or not.  The test fails if the the local server starts up, but the client session list remains at 0 for more than 60 seconds (as it usually responds within 6-12 seconds). When this happens I manually call SetHostInfo again and the client will then pick up this session.

So while I did reproduce this error once using an empty project and the latest version of bolt from the asset store, this project has since grown and I'm no longer using the BoltPhotonInit.cs script, but a modified version of it to make testing a little bit easier.

That being said, I replaced my PhotonPoller with your file linked above, and I read through your PhotonInit and made appropriate changes to my code that is identical to your connect code while still maintaining the automatic testing.  PhotonTokens was not needed for this test.  

Unfortuantely, after running my tests again, these changes did not fix the issue.  After 84 attempts the session list got stuck at 0, and only detected the local session after manually setting the host info a second time.

**Comment by DMeville** [comment]
Just ran a test x250 (start client, start server, server sets host info, client waits for it to show up) and it passed every time.  Looks like it might be fixed!

**Comment by ramonmelo** [comment]
Really nice the hear. So we can close this issue ? 

The fix will be on the next release.

**Comment by DMeville** [comment]
Yes, this appears fixed.  Let it run for a few hours and everything worked as expected. This can be closed. Thanks!
